libraries/auth/src/main/java/org/datatransferproject/auth/OAuth1Config.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2018 The Data Transfer Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.datatransferproject.auth;
+
+import com.google.api.client.auth.oauth.OAuthHmacSigner;
+import com.google.api.client.auth.oauth.OAuthSigner;
+import java.util.Map;
+
+public interface OAuth1Config {
+
+  /**
+   * Returns the name of the service, used for display, and client id and secret retrieval
+   */
+  String getServiceName();
+
+  /**
+   * Returns the request token url
+   */
+  String getRequestTokenUrl();
+
+  /**
+   * Returns the authorization url
+   */
+  String getAuthorizationUrl();
+
+  /**
+   * Returns the access token url
+   */
+  String getAccessTokenUrl();
+
+  /**
+   * Returns a map of scopes needed for export, keyed by data type (e.g., PHOTOS, CALENDAR) as
+   * defined in the auth data generator or elsewhere
+   */
+  Map<String, String> getExportScopes();
+
+  /**
+   * Returns a map of scopes needed for import, keyed by data type (e.g., PHOTOS, CALENDAR) as
+   * defined in the auth data generator or elsewhere
+   */
+  Map<String, String> getImportScopes();
+
+  /**
+   * Returns the parameter name for scopes
+   */
+  String getScopeParameterName();
+
+  /**
+   * Shows what step the scopes should be requested in
+   */
+  default OAuth1Step whenAddScopes() {
+    return OAuth1Step.AUTHORIZATION;
+  }
+
+  /**
+   * Returns the {@link OAuthSigner} for the initial token request
+   */
+  OAuthSigner getRequestTokenSigner(String clientSecret);
```

Comment:
all impls look the same, want to provide an interface default for now and we can always split it out later if needed? (I know you mentioned you were not sure if HMAC is always used for oauth1)

Reply:
Good idea, done.